Source/GameBaseFramework/Private/Camera/Modifiers/GBFCameraModifierJumpZ.cpp

```diff
@@ -0,0 +1,156 @@
+#include "Camera/Modifiers/GBFCameraModifierJumpZ.h"
+
+#include "GameFramework/CharacterMovementComponent.h"
+#include "GameFramework/SpringArmComponent.h"
+
+#include <Engine/Canvas.h>
+#include <GameFramework/Character.h>
+
+UGBFCameraModifierJumpZ::UGBFCameraModifierJumpZ() :
+    LandingTransitionTime( 0.5f ),
+    LandOnSameHeightCheckTolerance( 1.0f ),
+    DistanceFromLastGroundedPositionToResetModifier( 50.0f ),
+    CurrentState( EState::WaitingForJump ),
+    LastGroundedCameraZPosition( 0.0f ),
+    LastGroundedCharacterZPosition( 0.0f ),
+    CurrentCharacterZPosition( 0.0f ),
+    CurrentCameraZPosition( 0.0f ),
+    LerpStartCameraZPosition( 0.0f ),
+    LerpEndCameraZPosition( 0.0f ),
+    LandingTransitionRemainingTime( 0.0f )
+{
+}
+
+bool UGBFCameraModifierJumpZ::IsDisabled() const
+{
+    if ( Super::IsDisabled() )
+    {
+        return true;
+    }
+
+    return Cast< ACharacter >( GetViewTarget() ) == nullptr;
+}
+
+void UGBFCameraModifierJumpZ::ModifyCamera( float delta_time, FVector view_location, FRotator view_rotation, float fov, FVector & new_view_location, FRotator & new_view_rotation, float & new_fov )
+{
+    auto * character = Cast< ACharacter >( GetViewTarget() );
+
+    if ( character == nullptr )
+    {
+        return;
+    }
+
+    CurrentCharacterZPosition = character->GetActorLocation().Z;
+    auto * cmc = character->GetCharacterMovement();
+
+    switch ( CurrentState )
+    {
+        case EState::WaitingForJump:
```

Comment:
it is not possible to replace the enum by classes to get rid of the two switches? each state will have a method to evaluate if you need to change the current state or not, a method to get the needed values for the modifier and one for the displaydebug

Reply:
yes that's possible but this seems a bit overkill for such a simple task